portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,92 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+let root = document.documentElement;
+
+const secretText = document.querySelector("#secret-text"); 
+const mePic = document.querySelector("#me-pic"); 
+const tomiePic = document.querySelector("#tomie-pic"); 
+const mainHeading = document.querySelector("#main-heading"); 
+
+//fade in main text
+const mainText = document.querySelector(".main-text");
+window.addEventListener('load', e=> {
+    mainText.classList.add("fade-text-in"); 
+})
+
 /**
- * Adds a random greeting to the page.
+ * Keeps track of mouse and updates our special, circle pointer to
+ * follow cursor movement. 
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+const mouseCursor = document.querySelector(".cursor");
+window.addEventListener('mousemove', e=> {
+    mouseCursor.style.top = e.pageY + "px"; 
+    mouseCursor.style.left = e.pageX + "px"; 
+})
+
+/**
+ * Animates the mouse as it interacts with section headers. 
+ */
+secretMessages = ["...", "....", "Would you like to see something fun?"]; 
```

Comment:
This is a constant.

In general, you should never omit var/const/let in your variable declarations.  Back in the day, Internet Explorer used to self-destruct when you omitted it (and lots of federal agencies and banks were stuck on IE6 for ages and ages).  I don't know if that's still true but better not to find out the hard way :)